improve the scaling docs (#2038)
risingwavelabs/risingwave-docs: docs/deploy/k8s-cluster-scaling.md

## Scaling policies

> RisingWave supports adaptive parallelism and fixed parallelism for each streaming jobs, including materialized views, sinks and tables.

Reviewer: ```suggestion
RisingWave supports adaptive and fixed parallelism for each streaming job, including materialized view, sink, and table.
```

    ```

    When there are many streaming jobs running in the cluster, it’s recommended to use fixed parallelism to avoid overloading the system.

RisingWave distributes its computation across lightweight threads called "streaming actors," which run simultaneously on CPU cores. By spreading these streaming actors across cores, RisingWave achieves parallel computation, resulting in improved performance, scalability, and throughput.

In both scaling modes, streaming actors will redistribute across the cluster to maintain balanced workloads.

## Scale-out

Scale-out here refers to the process of adding more **compute nodes** to the cluster. For frontend nodes, you can simply scale out/in by adding more frontend nodes to the cluster, because they are stateless and can be automatically discovered by the meta nodes.

1. First, add more compute nodes with `kubectl`.

    ```bash
    # If you are using risingwave-operator
    kubectl apply -f <file-with-more-replicas>.yaml # or kubectl edit RisingWave/<name>

    # If you are not using risingwave-operator
    kubectl scale statefulset/risingwave-compute --replicas=<number-of-replicas>
    ```

>     Then wait until new compute nodes starts.

Reviewer: ```suggestion
    Then wait until new compute nodes start.
```

2. If you are using fixed parallelism, you may need to manually adjust the parallelism of the streaming jobs to utilize the new compute nodes. For adaptive parallelism, the system will automatically adjust the parallelism to utilize the new compute nodes.

## Scale-in

Scale-in here refers to the process of decreasing **compute nodes** from the cluster. By default, there's a 5-minute delay in scale-in operations. The delay is intentional to prevent unnecessary heavy recovery operations caused by transient failures like network jitters and CPU stalls. To manually trigger immediate scale-in, use the following statement:

> 1. Run following commands to unregister a compute node

Reviewer: ```suggestion
1. Run following commands to unregister a compute node.
```

    ```bash
    # Find out an worker id to unregister
    risingwave ctl meta cluster_info

    risingwave ctl meta unregister-worker {id}
    ```

    The `risingwave` command can be found at any node of the cluster. Recommend to use `kubectl exec` to login a pod and run the command. 

> 2. Decrase the number of compute nodes

Reviewer: ```suggestion
2. Decrease the number of compute nodes.
```
